velero-plugins/migimagestream/shared.go

```diff
@@ -29,11 +32,28 @@ func copyImage(src, dest string, sourceCtx, destinationCtx *types.SystemContext)
 	if err != nil {
 		return []byte{}, fmt.Errorf("Invalid destination name %s: %v", dest, err)
 	}
-	manifest, err := copy.Image(context.Background(), policyContext, destRef, srcRef, &copy.Options{
-		SourceCtx:      sourceCtx,
-		DestinationCtx: destinationCtx,
-	})
-	return manifest, err
+
+	// Let's retry the image copy up to 10 times
+	// Each retry will wait 5 seconds longer
+	// Let's log a warning if we encounter `blob unknown to registry`
+	retryWait := 5
+	log.Info(fmt.Sprintf("copying image: %s; will attempt up to 5 times...", src))
+	for i := 0; i < 4; i++ {
+		manifest, err := copy.Image(context.Background(), policyContext, destRef, srcRef, &copy.Options{
+			SourceCtx:      sourceCtx,
+			DestinationCtx: destinationCtx,
+		})
+		if err == nil {
+			return manifest, err
```

Comment:
Minor nit, but I'd return nil explicitly here to be clear we're returning no error.

Reply:
Done, thanks!